From ee97b3f18b4b95a1d92006410d6c6de921015ca1 Mon Sep 17 00:00:00 2001
From: kvandenm <koen.van-den-maagdenberg.xn@renesas.com>
Date: Thu, 23 Mar 2023 16:32:44 +0100
Subject: [PATCH] Modifications to SDK needed to build BLE SDK6 git example
 multi_con_periph. These modifications are picked from branch
 SDK6.1182.1_patch_multi_con_periph

---
 sdk/ble_stack/rwble/rwble.c            | 12 ++++++++++++
 sdk/platform/arch/arch.h               |  4 ++++
 sdk/platform/arch/main/arch_system.c   |  4 ++--
 sdk/platform/utilities/otp_cs/otp_cs.c |  2 +-
 4 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/sdk/ble_stack/rwble/rwble.c b/sdk/ble_stack/rwble/rwble.c
index 0793761..a500e7b 100644
--- a/sdk/ble_stack/rwble/rwble.c
+++ b/sdk/ble_stack/rwble/rwble.c
@@ -74,12 +74,20 @@
 #if (WLAN_COEX_ENABLED)
 #include "wlan_coex.h"
 #endif
+#include "lld_util.h"    
+uint8_t change_bd_addr_flag = 0; 
+uint8_t seed_region = 0x00;
+uint8_t store_adv_data_len                     __SECTION_ZERO("retention_mem_area0"); //@RETENTION MEMORY
+uint8_t test_scan_rsp_data_len                __SECTION_ZERO("retention_mem_area0"); //@RETENTION MEMORY
+uint8_t store_adv_data[ADV_DATA_LEN]           __SECTION_ZERO("retention_mem_area0"); //@RETENTION MEMORY
+uint8_t test_scan_rsp_data[SCAN_RSP_DATA_LEN] __SECTION_ZERO("retention_mem_area0"); //@RETENTION MEMORY
 
 last_ble_evt        arch_rwble_last_event           __SECTION_ZERO("retention_mem_area0");
 boost_overhead_st   set_boost_low_vbat1v_overhead   __SECTION_ZERO("retention_mem_area0");
 
 extern uint16_t clk_freq_trim_reg_value;
 
+
 #if defined (__DA14531__)
 extern uint16_t xtal_wait_trim;
 #endif
@@ -941,6 +949,9 @@ const ble_irq_t ble_irq[] =
 
 __BLEIRQ void rwble_isr(void)
 {
+	
+/* Optimize  the time between Tx pulses */
+ble_advtim_set (500);		
 #if defined (__DA14531__)
     if (GetWord32(BLE_CNTL2_REG) & EMACCERRSTAT)
     {
@@ -1091,6 +1102,7 @@ __BLEIRQ void rwble_isr(void)
             DBG_SWDIAG(BLE_ISR, ERRORINT, 0);
         }
     }
+
 #endif // __EXCLUDE_ROM_RWBLE__
 }
 
diff --git a/sdk/platform/arch/arch.h b/sdk/platform/arch/arch.h
index 4abf02a..0735065 100644
--- a/sdk/platform/arch/arch.h
+++ b/sdk/platform/arch/arch.h
@@ -115,6 +115,10 @@ extern uint32_t *rom_cfg_table;
 #define SWD_DATA_AT_P0_5    (0x01)
 #define SWD_DATA_AT_P0_1    (0x02)
 #define SWD_DATA_AT_P0_10   (0x03)
+#define HW_STARTUP_TIME_IN_XTAL32K_CYCLES    (11)   // 11 LP clocks for startup state machine handling
+#define HW_STARTUP_TIME_IN_RCX_CYCLES        (7)    // 7 LP clocks for startup state machine handling
+#define RCX_BLE_PWR_UP_TO_SLP_IRQ_USEC       (60)
+#define XTAL32K_BLE_PWR_UP_TO_SLP_IRQ_USEC   (45)
 #endif
 
 /** @brief OTP memory base address */
diff --git a/sdk/platform/arch/main/arch_system.c b/sdk/platform/arch/main/arch_system.c
index ba168af..4cd214a 100644
--- a/sdk/platform/arch/main/arch_system.c
+++ b/sdk/platform/arch/main/arch_system.c
@@ -662,7 +662,7 @@ void conditionally_run_radio_cals(void)
         last_temp_time = 0;
     }
 
-    if ((current_time - last_temp_time) >= 3200) //2 sec
+    if ((current_time - last_temp_time) >= 500000) 
     {
         last_temp_time = current_time;
 
@@ -958,7 +958,7 @@ uint8_t check_sys_startup_period(void)
 
 __WEAK bool app_use_lower_clocks_check(void)
 {
-    return false;
+    return true;
 }
 
 void lld_sleep_init_func(void)
diff --git a/sdk/platform/utilities/otp_cs/otp_cs.c b/sdk/platform/utilities/otp_cs/otp_cs.c
index 709fdcb..d29b0f7 100644
--- a/sdk/platform/utilities/otp_cs/otp_cs.c
+++ b/sdk/platform/utilities/otp_cs/otp_cs.c
@@ -37,7 +37,7 @@
  * entry.
  ****************************************************************************************
  */
-#define XTAL32M_WAIT_TRIM_TIME_USEC      (1000)  // 1msec
+#define XTAL32M_WAIT_TRIM_TIME_USEC      (500)  // 500usec
 
 /*
  * GLOBAL VARIABLES
-- 
2.39.2.windows.1

